rackcity/utils/execute_change_planner_utils.py

```diff
@@ -0,0 +1,117 @@
+from django.http import JsonResponse
+from http import HTTPStatus
+from rackcity.api.serializers import (
+    AddDecommissionedAssetSerializer,
+    RecursiveAssetSerializer,
+)
+from rackcity.models import (
+    Asset,
+    NetworkPort,
+    NetworkPortCP,
+    PDUPort,
+    PowerPort,
+    PowerPortCP,
+)
+from rackcity.utils.errors_utils import (
+    Status,
+    parse_serializer_errors,
+    parse_save_validation_error,
+)
+
+
+def get_updated_asset(asset_cp):
+    created = False
+    related_asset = asset_cp.related_asset
+    if related_asset:
+        updated_asset = related_asset
+    else:
+        updated_asset = Asset()
+        created = True
+    for field in Asset._meta.fields:
+        if field != 'id':
```

Comment:
Not sure if you tested this and there were no issues, but you might need to also ignore this field: `field.name == "assetid_ptr"` because this is how it stores it's reference to AssetID, which actually stores the id field

Reply:
i think when i left out `assetid_ptr` field i  accidentally overwrote the autofield,  resulting in two asset/assetCP with the same id.

Reply:
cool thx 4 flagging i will update